static/app/views/replays/replayTable/index.tsx

```diff
@@ -36,161 +39,190 @@ type Props = {
+                        return (
+                          <ActivityCell
+                            key="activity"
+                            replay={replay}
+                            showDropdownFilters={showDropdownFilters}
+                          />
+                        );
+
+                      case ReplayColumn.BROWSER:
+                        return (
+                          <BrowserCell
+                            key="browser"
+                            replay={replay}
+                            showDropdownFilters={showDropdownFilters}
+                          />
+                        );
+
+                      case ReplayColumn.COUNT_DEAD_CLICKS:
+                        return (
+                          <DeadClickCountCell
+                            key="countDeadClicks"
+                            replay={replay}
+                            showDropdownFilters={showDropdownFilters}
+                          />
+                        );
+
+                      case ReplayColumn.COUNT_ERRORS:
+                        return (
+                          <ErrorCountCell
+                            key="countErrors"
+                            replay={replay}
+                            showDropdownFilters={showDropdownFilters}
+                          />
+                        );
+
+                      case ReplayColumn.COUNT_RAGE_CLICKS:
+                        return (
+                          <RageClickCountCell
+                            key="countRageClicks"
+                            replay={replay}
+                            showDropdownFilters={showDropdownFilters}
+                          />
+                        );
+
+                      case ReplayColumn.DURATION:
+                        return (
+                          <DurationCell
+                            key="duration"
+                            replay={replay}
+                            showDropdownFilters={showDropdownFilters}
+                          />
+                        );
+
+                      case ReplayColumn.OS:
+                        return (
+                          <OSCell
+                            key="os"
+                            replay={replay}
+                            showDropdownFilters={showDropdownFilters}
+                          />
+                        );
+
+                      case ReplayColumn.REPLAY:
+                        return (
+                          <StyledReplayCell
+                            key="session"
+                            replay={replay}
+                            eventView={eventView}
+                            organization={organization}
+                            referrer={referrer}
+                            referrer_table="main"
+                            isPlaying={selectedReplayIndex === index}
+                            extraButton={buttonForReplayCell}
```

Comment:
We could make a new Cell type that is this button. Add an extra column to the table instead of messing with the existing cell and 2 ternaries. 

Instead of having to pass these props into the new Cell type, i'd put `selectedReplayIndex` into the url and let the Cell read it from there. can keep matching it to the new `index` props that comes from the map above.
The onClick could update the location, or call play/pause on the useReplayContext and whether the thing is selected, finally `replayPlayButtonPriority` is also is based on useReplayContext.

Reply:
for urls it would be even better to have the replay_id in there, instead of the list index, because the list index can change over time. but having anything in the url gets the link sharing setup right, and consolidates all the logic for this play/pause/change-selection button

Reply:
@ryan953 You don't think making a different replay cell will mean we have too much duplicated code?

Reply:
I'd do two distinct cells:
```
const PlayPauseCell = (props) => <Button ><IconPlay/></Button>;

const ReplayCell = // same as before
```